src/blocks/frontend/live-search/index.ts

```diff
@@ -0,0 +1,252 @@
+		}
+
+		const { height, fontSize } = getComputedStyle( inputElement );
+		const parentStyle = inputElement.parentElement ? getComputedStyle( inputElement.parentElement ) : null;
+
+		const container = document.createElement( 'div' );
+
+		container.classList.add( CONTAINER_CLASS );
+		container.style.width = inputElement.offsetWidth + 'px';
+		container.style.top = `calc( ${height} + ${parentStyle?.paddingTop} + ${parentStyle?.paddingBottom} + ${parentStyle?.borderBottomWidth} )`;
+		container.style.fontSize = `calc( ${fontSize} - 4px )`;
+
+		addLoadingIcon( container );
+		block?.appendChild( container );
+
+		return container;
+	};
+
+	const removeResultsContainer = ( block: Element | null, resultsContainer: ResultsContainer, cache = true ) => {
+		const tmpResultsContainer = block?.querySelector( `.${CONTAINER_CLASS}` );
+		if ( ! tmpResultsContainer ) {
+			return;
+		}
+
+		if ( cache ) {
+			return block?.removeChild( tmpResultsContainer as Node ) as ResultsContainer;
+		}
+
+		block?.removeChild( tmpResultsContainer as Node );
+		return null;
+	};
+
+	const updateResults = ( searchValue: string, block: Element | null, results: Array<ResultsEntry> ) => {
+		const container = block?.querySelector( `.${CONTAINER_CLASS}` );
+		if ( ! container ) {
+			return;
+		}
+
+		// Delete previous content
+		container.innerHTML = '';
+
+		if ( 0 === results.length ) {
+			const option = document.createElement( 'div' );
+
+			option.classList.add( `${CONTAINER_CLASS}__row`, 'no-results' );
+			option.innerHTML = strings.noResults + ` "<b>${ searchValue }</b>"`;
+
+			container?.appendChild( option );
+			return;
+		}
+
+		results.forEach( ({ link, title }, index ) => {
+			const option = document.createElement( 'div' );
+			const optionLink = document.createElement( 'a' );
+
+			option.classList.add( `${CONTAINER_CLASS}__row` );
+			( 0 === index ) && option.classList.add( 'highlight' );
+
+			optionLink.href = link;
+			optionLink.innerText = title;
+
+			option.appendChild( optionLink );
+			option.addEventListener( 'mouseover', () => {
+				const highlighted = container.querySelector( '.highlight' );
+				highlighted?.classList.remove( 'highlight' );
+				option.classList.add( 'highlight' );
+			});
+
+			container?.appendChild( option );
+		});
+	};
+
+	const addLoadingIcon = ( container: ResultsContainer ) => {
+		if ( ! container || container.querySelector( '.spinner-container' ) ) {
+			return;
+		}
+
+		const loading = document.createElement( 'div' );
+		loading.classList.add( `${CONTAINER_CLASS}__row`, 'spinner-container' );
+		loading.innerHTML = loadingIcon;
+
+		container.innerHTML = '';
+		container?.appendChild( loading );
+	};
+
+	const removeLoadingIcon = ( block: Element | null ) => {
+		const container = block?.querySelector( `.${CONTAINER_CLASS}` );
+		if ( ! container ) {
+			return;
+		}
+
+		const loading = container.querySelector( '.spinner-container' );
+		loading && container.removeChild( loading as Node );
```

Comment:
This also looks like an if with extra steps. The good old if is always reliable 👴 
```javascript
if( container.querySelector( '.spinner-container' ) ) {
    container.removeChild( loading as Node );
}

Reply:
It would look like this

``` javascript
if ( container.querySelector( '.spinner-container' ) ) {
	container.removeChild( container.querySelector( '.spinner-container' ) as Node );
}
```
or like this
``` javascript
const loading = container.querySelector( '.spinner-container' );
if ( loading ) {
	container.removeChild( loading as Node );
}
```

which one is better?

Reply:
`loading && container.removeChild( loading as Node )` is more like an expression. Using an expression as a control flow _only_ might not be very clear at first sight.